src/Sylius/Bundle/OrderBundle/Form/DataMapper/OrderItemQuantityDataMapper.php

```diff
@@ -60,8 +60,7 @@ public function mapFormsToData($forms, &$data)
 
         $formsOtherThanQuantity = [];
         foreach ($forms as $key => $form) {
-            if ('quantity' === $form->getName()) {
-                $targetQuantity = $form->getData();
+            if ('quantity' === $form->getName() && $targetQuantity = $form->getData()) {
```

Comment:
`OrderItemQuantityModifier::modify()` already handles target quantity of `0` with a softly typed check, so it shouldn't have a problem handling it. 

The only significant change is that if the `$targetQuantity == 0` you have now pushed it to the standard propertyPathDataMapper below instead. This can't be what we want, even if the side-effect is that it somehow prevents the exception...